app/models/application_record.rb

```diff
@@ -1,3 +1,5 @@
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+  scope :upcoming_date, -> { select { |event| event.date >= Time.now } }
+  scope :past_date, -> { select { |event| event.date < Time.now } }
```

Comment:
Most of the time, in **99.99%** of the case, we don't touch to this class.

Move it into `Event` model. 
Also, rewrite them like this
```rb
scope :upcoming_date, -> { where('date >= ?', Time.zone.now) }
```
Always think about using `Time.zone` instead of simply `Time`.